osh/hub/scripts/osh-retention

```diff
@@ -0,0 +1,99 @@
+
+
+def deleteScansResults(scansToDelete):
+    from kobo.hub import models
+    for scan in scansToDelete:
+        try:
+            pathToDelete = models.Task.get_task_dir(scan.id)
+            if os.path.isdir(pathToDelete):
+                shutil.rmtree(pathToDelete)
+        except OSError as e:
+            logging.error(f"Error deleting scans: {e}")
+
+
+# In order to identify personal scans, we are using the owner field. If the owner doesn't contain any / or @, we assume
+# that is a personal scan
+def identifyPersonalScan(owner):
+    # We could use regexps here if we want to make this more complex but for this specific case
+    # built-in string methods are enough and we are more perfomance-oriented.
+    if "/" in owner or "@" in owner:
+        return False
+    else:
+        return True
+
+
+def main():
+    import django
+    django.setup()
+
+    # Retention policy Scans
+    print("Starting Retention Policy Enforcement")
+
+    from osh.hub.scan import models
+    tasks = models.Task.objects.all()
+
+    # ---------- Section to take parameters from DB
+
+    retentionPersonalScans = 14
+    retentionFailedStatus = 14
+
+    if models.RetentionPolicySetting.objects.filter(category="PersonalScan").exists():
+        retentionPersonalObject = models.RetentionPolicySetting.objects.get(category="PersonalScan")
+        retentionPersonalScans = retentionPersonalObject.days
+    else:
+        # We set up the default time
+        retentionPersonalScans = 14
+        logging.warning("Error getting PersonalScan parameter. Setting default parameter")
+
+    if models.RetentionPolicySetting.objects.filter(category="FailedStatus").exists():
+        retentionFailedObject = models.RetentionPolicySetting.objects.get(category="FailedStatus")
+        retentionFailedStatus = retentionFailedObject.days
+    else:
+        # We set up the default time
+        retentionFailedStatus = 14
+        logging.warning("Error getting FailedStatus parameter. Setting default parameter")
+
+    # ---------- End Section to take parameters from DB
+
+    eligiblePersonalScans = []
+    eligibleFailedStatus = []
+
+    # We identify which scans are eligible to be deleted
+    today = date.today()
+    for t in tasks:
+        # At the moment, we only have 2 rule for retention settings:
+        # Personal Scans
+        # Failed status
+        calculatedDatePersonalScans = (t.dt_created + datetime.timedelta(days=retentionPersonalScans)).date()
+        calculatedDateStatusFailed = (t.dt_created + datetime.timedelta(days=retentionFailedStatus)).date()
+        if calculatedDatePersonalScans <= today:
+            if identifyPersonalScan(t.owner.username):
+                # Personal Scans eligible to be deleted
+                eligiblePersonalScans.append(t)
+        elif calculatedDateStatusFailed <= today:
+            # Failed status eligible to be deleted. Status 5 means to be deleted:
+            if t.state == 5:
```

Comment:
`5` is a magic number.  Please use the corresponding failed task state tuple from `kobo`: https://github.com/release-engineering/kobo/blob/986f2a305f9c675dbb54e34c91f09e443e328af3/kobo/client/constants.py#L36